src/test/compile-fail/empty-struct-with-braces-1.rs

```diff
@@ -8,13 +8,10 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-// compile-flags: -Z parse-only
+// Empty struct defined with braces shouldn't add names into value namespace
```

Comment:
This is a part of the RFC and a result of long discussion.
Personally, I find this quite logical - `struct S {}` is a usual struct and it doesn't add anything to the value namespace, it just happens to have an empty list of fields. `struct S;` in its turn is syntactically special and represent usual struct + some special magic.